app/models/user.rb

```diff
@@ -601,23 +597,15 @@ def create_personal_herbarium
   # Project that the User is a member of.
   def all_editable_species_lists(include: nil)
     @all_editable_species_lists ||= begin
-      if projects_member.any?
-        SpeciesList.includes(include).find_by_sql(
-          arel_select_species_lists_by_user_or_project.to_sql
-        )
-      else
-        species_lists.includes(include)
-      end
-    end
-  end
+      return species_lists.includes(include) if projects_member.none?
```

Comment:
I am concerned that this will avoid setting @all_editable_species_lists, defeating the memoization.  While I hate unnecessarily splitting methods up, this may be a "necessary" case.  Consider moving the body into something like "calc_all_editable_species_lists", and then make all_editable_species_lists a trivial memoized method: @all_editable_species_lists ||= calc_all_editable_species_lists.  Then when your guard clause here returns early, it will still save the result in @all_editable_species_lists and not have to run species_lists.includes(include) multiple times if all_editable_species_lists is called multiple times.